### ISSUE TITLE
Remove usage of microdom

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7943]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#7943|
|Type|enhancement|
|Created|2015-06-17 09:57:51Z|
|Branch|https://github.com/twisted/twisted/tree/remove-microdom-usage-7943|

In order to deprecate microdom as requested in [#3561](https://github.com/twisted/twisted/issues/3561) we first need to remove all usage of microdom from Twisted code.

<details><summary>Searchable metadata</summary>

```
trac-id__7943 7943
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__ 
branch__branches_remove_microdom_usage_7943 branches/remove-microdom-usage-7943
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1434535071015376 1434535071015376
changetime__1452337674102803 1452337674102803
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

Luckily there's not a whole lot at this point.

  - domhelpers, which should be deprecated right along with `microdom`,
  - some unused imports (!?!?!) in the finger tutorial,
  - a disused example, `lj.rpy.py`,
  - one use, which could probably just be replaced with an "is this file empty" check, that is just making sure Sphinx produced some HTML, in `test_release`.

I think that's it.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

(In [46567]) Branching to remove-microdom-usage-7943.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

Thanks glyph for the checklist :)

* domhelpers was not used
* Removed unused microdom imports from the finger tutorial
* Removed lj.rpy and its reference
* test_release.py was already updated but the docs were not updated. I have updated them now.

--------

Not sure if sux is a problem, or only microdom is broken. I have never used Twisted with jabber.

sux is still used in twisted/words/xish/domish.py which is used by twisted/words/protocols/jabber/

----

Please review. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @adiroiban|
|-|-|

This looks fairly reasonable (glad to see there were so few uses remaining!), but the tests are failing. Please fix and resubmit?

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban removed owner|
|-|-|

there was an accidental character at the start of the file... my bad.

hope all is fine now. sent to buildbot.

please review. thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set owner to @hawkowl|
|-|-|

Hi Adi, looks good. Will merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46600]) Merge remove-microdom-usage-7943: Remove usage of microdom

Author: adiroiban
Reviewers: glyph, hawkowl
Fixes: [#7943](https://github.com/twisted/twisted/issues/7943)
